changelog.d/pr-7002.md

```diff
@@ -0,0 +1,4 @@
+### Tests
+
+- Fix broken test helpers for result record testing that would falsely pass.
+  https://github.com/datalad/datalad/pull/7002 (by @bpoldrack)
```

Comment:
I think we should tune up our template ... filed #7084 . up to you to fix it up or not here.

Reply:
You mean like this: https://github.com/datalad/datalad/pull/7002/commits/4ce17b19d7a317d215c59711aedc78658bdcfb4e ?